src/main/java/uk/gov/pay/adminusers/service/ServiceInviteCreator.java

```diff
@@ -61,35 +63,48 @@ public Invite doCreate(InviteServiceRequest inviteServiceRequest) {
         if (inviteOptional.isPresent()) {
             InviteEntity foundInvite = inviteOptional.get();
             if (!foundInvite.isExpired() && !foundInvite.isDisabled()) {
-                sendUserInviteNotification(foundInvite);
-                throw conflictingInvite(requestEmail);
+                if (USER.getType().equals(foundInvite.getType())) {
+                    sendUserInviteNotification(foundInvite);
+                    throw conflictingInvite(requestEmail);
+                } else {
+                    return constructInviteAndSendEmail(inviteServiceRequest, foundInvite, true);
+                }
             }
         }
 
         return roleDao.findByRoleName(inviteServiceRequest.getRoleName())
                 .map(roleEntity -> {
                     InviteEntity inviteEntity = new InviteEntity(requestEmail, randomUuid(), inviteServiceRequest.getOtpKey(), null, null, roleEntity);
-                    inviteEntity.setTelephoneNumber(inviteServiceRequest.getTelephoneNumber());
-                    inviteEntity.setType(SERVICE);
-                    inviteDao.persist(inviteEntity);
-                    String inviteUrl = format("%s/%s", linksConfig.getSelfserviceInvitesUrl(), inviteEntity.getCode());
-                    sendInviteNotification(inviteEntity, inviteUrl);
-                    return linksBuilder.decorate(inviteEntity.toInvite(inviteUrl));
+                    return constructInviteAndSendEmail(inviteServiceRequest, inviteEntity, false);
                 })
                 .orElseThrow(() -> internalServerError(format("Role [%s] not a valid role for creating a invite service request", inviteServiceRequest.getRoleName())));
 
     }
 
-    private void sendUserInviteNotification(InviteEntity invite) {
-        String inviteUrl = fromUri(linksConfig.getSelfserviceInvitesUrl()).path(invite.getCode()).build().toString();
-        UserEntity sender = invite.getSender();
-        notificationService.sendInviteEmail(sender.getEmail(), invite.getEmail(), inviteUrl)
-                .thenAcceptAsync(notificationId -> LOGGER.info("invite resent successfully to user, notification id [{}]", notificationId))
+    private Invite constructInviteAndSendEmail(InviteServiceRequest inviteServiceRequest, InviteEntity inviteEntity, boolean toUpdate) {
+        String inviteUrl = format("%s/%s", linksConfig.getSelfserviceInvitesUrl(), inviteEntity.getCode());
+        inviteEntity.setTelephoneNumber(inviteServiceRequest.getTelephoneNumber());
+        inviteEntity.setPassword(passwordHasher.hash(inviteServiceRequest.getPassword()));
+        if (toUpdate) {
```

Comment:
I find this condition a bit awkward, this method is doing two different things depending on a flag. Something to consider for refactoring in next PR?